apollo-router/src/configuration/cors.rs

```diff
@@ -44,22 +151,17 @@ pub(crate) struct Cors {
     /// in response to a cross-origin request.
     pub(crate) expose_headers: Option<Vec<String>>,
 
-    /// The origin(s) to allow requests from.
-    /// Defaults to `https://studio.apollographql.com/` for Apollo Studio.
-    pub(crate) origins: Vec<String>,
-
-    /// `Regex`es you want to match the origins against to determine if they're allowed.
-    /// Defaults to an empty list.
-    /// Note that `origins` will be evaluated before `match_origins`
-    pub(crate) match_origins: Option<Vec<String>>,
-
-    /// Allowed request methods. Defaults to GET, POST, OPTIONS.
+    /// Allowed request methods. See module documentation for default behavior.
     pub(crate) methods: Vec<String>,
```

Comment:
As a consumer consumer of this feature, this is not currently working as I would expect.

When:
- this global default "methods" is set to POST in the top level config
- there exists policy/policies that do not specify "methods"
- at runtime origins that match the policies are currently getting response of GET, POST, OPTIONS
- this defeats the whole purpose of having a global default

Reply:
Verified this is now working as I would expect.